superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -124,6 +205,145 @@ export default class ResultSet extends React.PureComponent<
     }
   }
 
+  handleOnChangeAutoComplete = () => {
+    this.setState({ datasetToOverwrite: {} });
+  };
+
+  handleOverwriteDataset = async () => {
+    const { sql, results, dbId } = this.props.query;
+    const { datasetToOverwrite } = this.state;
+
+    updateDatset(
+      datasetToOverwrite.datasetId,
+      dbId,
+      sql,
+      results.selected_columns.map(d => ({ column_name: d.name })),
+      true,
+    )
+      .then(() => {
+        exploreChart({
+          ...EXPLORE_CHART_DEFAULT,
+          datasource: `${datasetToOverwrite.datasetId}__table`,
+          all_columns: results.selected_columns.map(d => d.name),
+        });
+      })
+      .catch(() => {
+        this.props.actions.addDangerToast(
+          t('An error occurred overwriting dataset'),
+        );
+      });
+
+    this.setState({
+      showSaveDatasetModal: false,
+      shouldOverwriteDataSet: false,
+      datasetToOverwrite: {},
+      newSaveDatasetName: `${this.props.query.tab} ${moment().format(
+        'MM/DD/YYYY HH:mm:ss',
```

Comment:
This is US English only. No need to resolve in this PR, but we should really start thinking about an i18n strategy for things like date formatting. 

cc @villebro @rusackas

Reply:
+1

Reply:
"YYYY-MM-DD HH:mm:ss" It's a more common way for i18n.

Reply:
+1 on using `YYYY-MM-DD HH:mm:ss` by default before we add proper internalization/localization support.